### PR TITLE
SALTO-6667 Zendesk trigger skill priority

### PR DESCRIPTION
**Pull request by adfineberg**
Support a new API format of skills within Zendesk triggers.

---

_Additional context for reviewer_

---
_Release Notes_: 
_Zendesk Adapter:_
* set_skills or add_skills under Trigger actions will now have a priority of required or optional as per https://developer.zendesk.com/documentation/ticketing/using-the-zendesk-api/setting-skill-priority-with-skills-in-trigger-action/

---
_User Notifications_: 
_Zendesk Adapter:_
* Trigger nacls with set_skills or add_skills will now include a priority key if this is received from Zendesk.

### CODE REVIEW

File: packages/zendesk-adapter/src/filters/deploy_trigger_skills.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  instance: InstanceElement,
+  skillMapping: Record<string, { value: string; priority: string }>,
+): Promise<void> => {
+  instance.value?.actions
+    .filter((action: unknown) => TRIGGER_SKILL_FIELDS.includes(_.get(action, 'field')))
+    .forEach((action: { value?: string; priority?: string }) => {
+      if ('priority' in action && 'value' in action) {
+        const { value, priority } = action as { value: string; priority: string }
+        action.value = `${value}#${priority === 'optional' ? '1' : '0'}`
+        skillMapping[action.value] = { value, priority }
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/deploy_trigger_skills.ts:24]: can you have duplicates here? meaning that `action.value` will appear twice?

File: packages/zendesk-adapter/src/filters/deploy_trigger_skills.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          .filter(instance => instance.elemID.typeName === TRIGGER_TYPE_NAME)
+          .map(instance => restoreTriggerSkillToApi(instance, skillMapping)),
+      )
+    },
+    onDeploy: async (changes: Change<InstanceElement>[]): Promise<void> =>
+      changes
+        .map(getChangeData)
+        .filter(instance => instance.elemID.typeName === TRIGGER_TYPE_NAME && Array.isArray(instance.value?.actions))
+        .forEach(async instance => {
+          instance.value.actions = instance.value.actions.map((action: { value: string }) => {
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/filters/deploy_trigger_skills.ts:51]: can you check that action is of the correct type? that it has a value?

**Review by edenhassid — COMMENTED** [review]
looks good! are we sure that this is only possible for triggers and no other types?

File: packages/zendesk-adapter/src/filters/deploy_trigger_skills.ts (hunk 1 above)
Comment (adfineberg) [reply packages/zendesk-adapter/src/filters/deploy_trigger_skills.ts:24]: It's ok, they're always the same mapping

**Review by edenhassid — APPROVED** [review]

**Pull request merged**